src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/InstanceDiscovery.cs

```diff
@@ -0,0 +1,151 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    [DataContract]
+    internal sealed class InstanceDiscoveryResponse
+    {
+        [DataMember(Name = "tenant_discovery_endpoint")]
+        public string TenantDiscoveryEndpoint { get; set; }
+
+        [DataMember(Name = "metadata")]
+        public InstanceDiscoveryMetadataEntry[] Metadata { get; set; }
+    }
+
+    [DataContract]
+    internal sealed class InstanceDiscoveryMetadataEntry
+    {
+        [DataMember(Name = "preferred_network")]
+        public string PreferredNetwork { get; set; }
+
+        [DataMember(Name = "preferred_cache")]
+        public string PreferredCache { get; set; }
+
+        [DataMember(Name = "aliases")]
+        public string[] Aliases { get; set; }
+    }
+
+    internal class InstanceDiscovery
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a 'protected' constructor or the 'static' keyword to the class declaration. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://iddp-sonarqube:9000/coding_rules#rule_key=csharpsquid%3AS1118)